Dockerfile

```diff
@@ -1,56 +1,162 @@
+WORKDIR /openms-streamlit-vue-component
+RUN npm install
+RUN npm run build
+
+FROM ubuntu:22.04 AS setup-build-system
+WORKDIR /
+
+COPY --from=js-build openms-streamlit-vue-component/dist /app/js-component/dist
+
+ARG OPENMS_REPO=https://github.com/t0mdavid-m/OpenMS.git
+ARG OPENMS_BRANCH=TaggerViewerDeployment
 ARG PORT=8501
 # GitHub token to download latest OpenMS executable for Windows from Github action artifact.
 ARG GITHUB_TOKEN
 # Streamlit app Gihub user name (to download artifact from).
-ARG GITHUB_USER=OpenMS
+ARG GITHUB_USER=t0mdavid-m
 # Streamlit app Gihub repository name (to download artifact from).
-ARG GITHUB_REPO=streamlit-template
+ARG GITHUB_REPO=FLASHTaggerViewer 
+
 
-# Step 1: set up a sane build system
 USER root
 
+# Install required Ubuntu packages.
 RUN apt-get -y update
-# note: streamlit in docker needs libgtk2.0-dev (see https://yugdamor.medium.com/importerror-libgthread-2-0-so-0-cannot-open-shared-object-file-no-such-file-or-directory-895b94a7827b)
-RUN apt-get install -y --no-install-recommends --no-install-suggests wget ca-certificates libgtk2.0-dev curl jq cron
+RUN apt-get install -y --no-install-recommends --no-install-suggests g++ autoconf automake patch libtool make git gpg wget ca-certificates curl jq libgtk2.0-dev openjdk-8-jdk cron
 RUN update-ca-certificates
+RUN apt-get install -y --no-install-recommends --no-install-suggests libsvm-dev libeigen3-dev coinor-libcbc-dev libglpk-dev libzip-dev zlib1g-dev libxerces-c-dev libbz2-dev libomp-dev libhdf5-dev
+RUN apt-get install -y --no-install-recommends --no-install-suggests libboost-date-time1.74-dev \
+                                                                     libboost-iostreams1.74-dev \
+                                                                     libboost-regex1.74-dev \
+                                                                     libboost-math1.74-dev \
+                                                                     libboost-random1.74-dev
+RUN apt-get install -y --no-install-recommends --no-install-suggests qtbase5-dev libqt5svg5-dev libqt5opengl5-dev
 
-#################################### install streamlit
-# install packages
-COPY requirements.txt requirements.txt
-RUN python -m pip install --upgrade pip
-RUN python -m pip install -r requirements.txt
+# Download and install mamba.
+ENV PATH="/root/mambaforge/bin:${PATH}"
+RUN wget -q \
+    https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh \
+    && bash Mambaforge-Linux-x86_64.sh -b \
+    && rm -f Mambaforge-Linux-x86_64.sh
+RUN mamba --version
+
+# Setup mamba environment.
+COPY environment.yml ./environment.yml
+RUN mamba env create -f environment.yml
+RUN echo "mamba activate streamlit-env" >> ~/.bashrc
+SHELL ["/bin/bash", "--rcfile", "~/.bashrc"]
+SHELL ["mamba", "run", "-n", "streamlit-env", "/bin/bash", "-c"]
+
+# Install up-to-date cmake via mamba and packages for pyOpenMS build.
```

Comment:
Do we have any reason to use mamba here?
@drewbudd removed mamba to reduce the size of the docker image

Reply:
I suppose you could get it to work using pip. However, I think this is something that should be adjusted in the [template repository](https://github.com/OpenMS/streamlit-template). Also note that we have to build the TOPP tools in the new version.